psc-ide.el

```diff
@@ -360,44 +370,59 @@ use when the search used was with `string-match'."
 
 
 (defun psc-ide-ask-project-dir ()
-  "Ask psc-ide-server for the project dir."
+  "Ask psc-ide server for the project dir."
   (interactive)
   (psc-ide-send psc-ide-command-cwd
                 (-compose 'message 'psc-ide-unwrap-result)))
 
 (defun psc-ide-server-start-impl (dir-name)
-  "Start psc-ide-server."
+  "Start psc-ide server."
   (apply 'start-process `("*psc-ide-server*" "*psc-ide-server*"
                           ,@(psc-ide-server-command dir-name))))
 
 (defun psc-ide-server-command (dir-name)
-  "Tries to find the psc-ide-server-executable and builds up the
+  "Tries to find the psc-ide executable and builds up the
   command by appending eventual options. Returns a list that can
   be expanded and passed to start-process"
-  (let* ((npm-bin-path (if psc-ide-use-npm-bin
-                           (psc-ide-npm-bin-server-executable)
+  (let* ((executable-name (psc-ide-executable-name))
+         (npm-bin-path (if psc-ide-use-npm-bin
+                           (psc-ide-npm-bin-server-executable executable-name)
                          nil))
-         (path (or npm-bin-path (executable-find psc-ide-server-executable)))
+         (path (or npm-bin-path (executable-find executable-name)))
+         (cmd (if psc-ide-use-purs
+                  `(,path "ide" "server")
+                `(,path)))
          (port (number-to-string psc-ide-port))
          (directory (expand-file-name dir-name))
-         (debug-flag (when psc-ide-debug "--debug"))
-         (globs (when (psc-ide--version-gte (psc-ide-server-version) "0.9.2") psc-ide-source-globs)))
+         (debug-flags (when psc-ide-debug (if psc-ide-use-purs
+                                              '("--log-level" "debug")
+                                            '("--debug"))))
+         (globs (when (and (not psc-ide-use-purs)
```

Comment:
Actually let's also default to `t` for the customize variable, we should encourage using the newest version of the compiler.